lib/srv/discovery/discovery.go

```diff
@@ -125,6 +125,11 @@ type Config struct {
 	// discovery cycles.
 	PollInterval time.Duration
 
+	// TriggerFetchC is a list of channels that must be notified when a off-band poll must be performed.
+	// This is used to start a polling iteration when a new DiscoveryConfig change is received.
+	TriggerFetchC  []chan struct{}
+	triggerFetchMu *sync.RWMutex
```

Comment:
Nit: I don't think your mutex needs to be a pointer, as long as you're using it on pointer receiver methods.